lib/segment/src/index/field_index/field_index_base.rs

```diff
@@ -114,6 +114,7 @@ pub enum FieldIndex {
     GeoIndex(GeoMapIndex),
     FullTextIndex(FullTextIndex),
     BinaryIndex(BinaryIndex),
+    UuidIndex(NumericIndex<UuidPayloadKeyType, UuidPayloadType>),
```

Comment:
After digging down into the encoding part I understood where the `Key` name came from, but in my mind it was conflicting with `PayloadKeyType`. Maybe a better name would be `UuidIntType`/`IntUuidType`?